GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java

```diff
@@ -236,6 +237,7 @@ protected void onDestroy() {
         mRenderableViewGroup = null;
         mConfigurationManager = null;
 
+        GVRJavascriptV8File.destroy();
```

Comment:
If a second activity is created for this process, the old activity's onDestroy is usually called after onCreate for the new one. Meaning there is theoretically a window of time where the incorrect instance could be used by the new activity.

Ideally there should be no static writable data but if you must, you could use GVRContext.addResetOnRestartHandler to set mEngine to null in a safer manner.

Reply:
OK will try this approach.

Reply:
Done

Reply:
you no longer need these changes to GVRActivity.  the ResetOnRestart handler is all that is needed.  in fact, since that destroy() method no longer exists, it now breaks the build.

Reply:
That's right. GVRActivity is not touched at all in my 2nd commit. However I am confused by the 2 commits appearing in this pull request. I used 'git commit --amend -a' before 'git push'. Did I miss something?

Reply:
Only the 2nd commit shall be merged. The 1st one shall be discarded. I may have used git commands incorrectly.